pkg/sdk/helper.go

```diff
@@ -129,3 +130,25 @@ func (h *Helper) ModelAndDocsPath(serviceModelName string) string {
 	)
 	return modelPath
 }
+
+// GetOriginalAPIShapeSet returns a map set (keys with empty struct values) of all original shape names
+// from the AWS service API model. This provides efficient lookups to check if a shape name exists.
+func (h *Helper) GetOriginalAPIShapeSet(serviceNameOrID string) (map[string]struct{}, error) {
+	modelPath := h.ModelAndDocsPath(serviceNameOrID)
+	apis, err := apiv2.LoadAPI(modelPath)
```

Comment:
i think one of the reasons why the code-generator is slower today is the increased runtime complexity of the apiv2.LoadApi. Not sure if we should take this route